consensus/service/src/api/client_api_service.rs

```diff
@@ -274,13 +292,54 @@ impl ConsensusClientApi for ClientApiService {
     ) {
         let _timer = SVC_COUNTERS.req(&ctx);
 
+        let session_id = ClientSession::from(msg.channel_id.clone());
+
         {
-            let session = ClientSession::from(msg.channel_id.clone());
             let mut tracker = self.tracked_sessions.lock().expect("Mutex poisoned");
             // Calling get() on the LRU bumps the entry to show up as more
             // recently-used.
-            if tracker.get(&session).is_none() {
-                tracker.put(session, ClientSessionTracking::new());
+            if tracker.get(&session_id).is_none() {
```

Comment:
:wrench: This feels like this block could be its own method similar to the `check_request_chain_id()` guard clause below